kratos/utilities/variable_utils.cpp

```diff
@@ -486,4 +487,94 @@ void VariableUtils::AuxiliaryAtomicAdd(
+        (IsInverseWeightProvided) ? r_inverse_method : r_direct_method;
+
+#pragma omp parallel for
+    for (int i_condition = 0; i_condition < number_of_conditions; ++i_condition)
+    {
+        auto& r_entity = *(r_entities.begin() + i_condition);
+        auto& r_geometry = r_entity.GetGeometry();
+
+        const auto& r_value = r_entity.GetValue(rVariable);
+        for (int i_node = 0; i_node < static_cast<int>(r_geometry.PointsNumber()); ++i_node)
+        {
+            auto& r_node = r_geometry[i_node];
+
+            KRATOS_DEBUG_ERROR_IF(!r_node.Has(rWeightVariable))
+                << "Non-historical nodal " << rWeightVariable.Name() << " at "
+                << r_node << " is not initialized in " << rModelPart.Name()
+                << ". Please initialize it first.";
+
+            const auto& r_current_value = r_node.GetValue(rVariable);
+            const double weight = r_weight_method(r_node);
+
+            r_node.SetLock();
+            r_node.SetValue(rVariable, r_current_value + r_value * weight);
+            r_node.UnSetLock();
+        }
+    }
+
+    rModelPart.GetCommunicator().AssembleNonHistoricalData(rVariable);
+
+    KRATOS_CATCH("");
+}
+
+// template instantiations
+template KRATOS_API(KRATOS_CORE) void VariableUtils::DistributeVariable<double, ModelPart::ConditionsContainerType, int>(
+    ModelPart&, const Variable<double>&, const Variable<int>&, const bool);
+template KRATOS_API(KRATOS_CORE) void VariableUtils::DistributeVariable<array_1d<double, 3>, ModelPart::ConditionsContainerType, int>(
+    ModelPart&, const Variable<array_1d<double, 3>>&, const Variable<int>&, const bool);
+
+template KRATOS_API(KRATOS_CORE) void VariableUtils::DistributeVariable<double, ModelPart::ElementsContainerType, int>(
+    ModelPart&, const Variable<double>&, const Variable<int>&, const bool);
+template KRATOS_API(KRATOS_CORE) void VariableUtils::DistributeVariable<array_1d<double, 3>, ModelPart::ElementsContainerType, int>(
+    ModelPart&, const Variable<array_1d<double, 3>>&, const Variable<int>&, const bool);
+
+template KRATOS_API(KRATOS_CORE) void VariableUtils::DistributeVariable<double, ModelPart::ConditionsContainerType, double>(
+    ModelPart&, const Variable<double>&, const Variable<double>&, const bool);
+template KRATOS_API(KRATOS_CORE) void VariableUtils::DistributeVariable<array_1d<double, 3>, ModelPart::ConditionsContainerType, double>(
+    ModelPart&, const Variable<array_1d<double, 3>>&, const Variable<double>&, const bool);
+
+template KRATOS_API(KRATOS_CORE) void VariableUtils::DistributeVariable<double, ModelPart::ElementsContainerType, double>(
+    ModelPart&, const Variable<double>&, const Variable<double>&, const bool);
+template KRATOS_API(KRATOS_CORE) void VariableUtils::DistributeVariable<array_1d<double, 3>, ModelPart::ElementsContainerType, double>(
+    ModelPart&, const Variable<array_1d<double, 3>>&, const Variable<double>&, const bool);
+
```

Comment:
I think that this can be avoided by using the container as argument (instead of the model part)

Reply:
Only problem is, If I am to use container as an argument, I again need the model part as well. Because,

First I need to clear the output variable of nodes to have zeros.
Secondly, I need the MPI communicator.

Thats why I created the GetContainer method, so everything can be obtained from model part.